android/src/main/java/com/equimaps/capacitor_background_geolocation/BackgroundGeolocation.java

```diff
@@ -76,90 +82,106 @@ public void addWatcher(final PluginCall call) {
             call.reject("Service not running.");
             return;
         }
-        call.setKeepAlive(true);
+        if (getPermissionState("location") != PermissionState.GRANTED && !call.getBoolean("requestPermissions", true)) {
+            call.reject("Permission denied.", "NOT_AUTHORIZED");
+            return;
+        }
 
-        if (getPermissionState("location") != PermissionState.GRANTED) {
-            if (call.getBoolean("requestPermissions", true)) {
-                requestPermissionForAlias("location", call, "locationPermissionsCallback");
-            } else {
-                call.reject("Permission denied.", "NOT_AUTHORIZED");
-            }
```

Comment:
I think these nested `if` blocks are easier to understand. Why did you change it?